test_conformance/basic/test_readimage3d.cpp

```diff
@@ -110,123 +108,123 @@ prepare_reference(unsigned char * input_ptr, int w, int h, int d)
+	cl_sampler sampler;
+
+	static testFormat formatsToTest[] =
+	{
+		{
+			"test_bgra8888",
+			bgra8888_kernel_code,
+			{CL_BGRA, CL_UNORM_INT8},
+		},
+		{
+			"test_rgba8888",
+			rgba8888_kernel_code,
+			{CL_RGBA, CL_UNORM_INT8},
+		},
+	};
+
+	unsigned char *input_ptr;
+	float *output_ptr;
+	double *ref_ptr;
+	size_t threads[3];
+	int img_width = 64;
+	int img_height = 64;
+	int img_depth = 64;
+	int i, err;
+	size_t origin[3] = {0, 0, 0};
+	size_t region[3] = {img_width, img_height, img_depth};
+	size_t length = img_width * img_height * img_depth * 4 * sizeof(float);
+
+	PASSIVE_REQUIRE_3D_IMAGE_SUPPORT( device )
+
+	for (uint32_t i = 0; i < ARRAY_SIZE(formatsToTest); i++)
+	{
+		if (!is_image_format_required(formatsToTest[i].img_format, CL_MEM_READ_ONLY, CL_MEM_OBJECT_IMAGE3D, device))
+			continue;
+
+		MTdata d = init_genrand( gRandomSeed );
+		input_ptr = generate_3d_image8(img_width, img_height, img_depth, d);
+		ref_ptr = prepare_reference(input_ptr, img_width, img_height, img_depth);
+		output_ptr = (float*)malloc(length);
+
+		streams[0] = create_image_3d(context, CL_MEM_READ_ONLY, &formatsToTest[i].img_format, img_width, img_height, img_depth, 0, 0, NULL, &err);
+		test_error(err, "create_image_3d failed");
+
+		streams[1] = clCreateBuffer(context, CL_MEM_READ_WRITE, length, NULL, &err);
+		test_error(err, "clCreateBuffer failed");
+
+		sampler = clCreateSampler(context, CL_FALSE, CL_ADDRESS_CLAMP_TO_EDGE, CL_FILTER_NEAREST, &err);
+		test_error(err, "clCreateSampler failed");
+
+		err = clEnqueueWriteImage(queue, streams[0], CL_TRUE, origin, region, 0, 0, input_ptr, 0, NULL, NULL);
+		test_error(err, "clEnqueueWriteImage failed");
+
+		err = create_single_kernel_helper(context, &program, &kernel, 1, &formatsToTest[i].kernelSourceString, formatsToTest[i].kernelName);
+		test_error(err, "create_single_kernel_helper failed");
+
+		err = clSetKernelArg(kernel, 0, sizeof streams[0], &streams[0]);
+		err |= clSetKernelArg(kernel, 1, sizeof streams[1], &streams[1]);
+		err |= clSetKernelArg(kernel, 2, sizeof sampler, &sampler);
+		test_error(err, "clSetKernelArg failed");
+
+		threads[0] = (unsigned int)img_width;
+		threads[1] = (unsigned int)img_height;
+		threads[2] = (unsigned int)img_depth;
+
+		err = clEnqueueNDRangeKernel(queue, kernel, 3, NULL, threads, NULL, 0, NULL, NULL);
+		test_error(err, "clEnqueueNDRangeKernel failed");
+
+		err = clEnqueueReadBuffer(queue, streams[1], CL_TRUE, 0, length, output_ptr, 0, NULL, NULL);
+		test_error(err, "clEnqueueReadBuffer failed");
+
+		err = verify_3d_image8(ref_ptr, output_ptr, img_width, img_height, img_depth);
+		if ( err == 0 )
+		{
+			switch (i)
+			{
+				case 0:
+				{
+					log_info("READ_IMAGE3D_BGRA_UNORM_INT8 test passed\n");
+					break;
+				}
+				case 1:
+				{
+					log_info("READ_IMAGE3D_RGBA_UNORM_INT8 test passed\n");
+					break;
+				}
+				default:
+				{
+					err = CL_OUT_OF_RESOURCES;
+					test_error(err, "Unhandled case");
+					break;
+				}
+			}
+		}
```

Comment:
It's a shame to have a big switch just to print a pass message with the format name. We probably have a format to string function somewhere.